software/contrib/harmonic_lfos.py

```diff
@@ -5,33 +5,49 @@
 from random import randint
 from europi_script import EuroPiScript
 
-MAX_VOLTAGE = MAX_OUTPUT_VOLTAGE #Default is inherited but this can be overriden by replacing "MAX_OUTPUT_VOLTAGE" with an integer
-MAX_HARMONIC = 32 #Too high a value may be hard to select using the knob, but the actual hardware limit is only reached at 4096
+MAX_VOLTAGE = MAX_OUTPUT_VOLTAGE # Default is inherited but this can be overriden by replacing "MAX_OUTPUT_VOLTAGE" with an integer
+MAX_HARMONIC = 32 # Too high a value may be hard to select using the knob, but the actual hardware limit is only reached at 4096
+
 
 class HarmonicLFOs(EuroPiScript):
+    MODES_SHAPES = {
+        'SINE': 0,
+        'SAW': 1,
+        'SQUARE': 2,
+        'OFF': 3,
+        'RANDOM': 4,
+        'NOISE': 5,
+    }
+
     def __init__(self):
         super().__init__()
         
-        #Retreive saved state information from file
+        # Retreive saved state information from file
         state = self.load_state_json()
         
-        #Overclock the Pico for improved performance
+        # Overclock the Pico for improved performance
         freq(250_000_000)
         
-        #Use the saved values for the LFO divisions and mode if found in the save state file, using defaults if not
+        # Use the saved values for the LFO divisions and mode if found in the save state file, using defaults if not
         self.divisions = state.get("divisions", [1, 3, 5, 7, 11, 13])
-        self.modes = state.get("modes", [0, 0, 0, 0, 0, 0])
-        self.MODE_COUNT = 5
-        
-        #Initialise all the other variables
+        self.modes = state.get("modes", [self.MODES_SHAPES['SINE']] * 6)
+        self.MODES_COUNT = len(self.MODES_SHAPES)
+
+        # Initialise all the other variables
         self.degree = 0
+        self.rad = radians(self.degree)
         self.delay, self.increment_value = self.get_delay_increment_value()
         self.pixel_x = OLED_WIDTH-1
         self.pixel_y = OLED_HEIGHT-1
         self.selected_lfo = 0
-        self.selected_lfo_start_value = self.get_clock_division()
-        
-        #Set the digital input and button handlers
+        self.clock_division = self.selected_lfo_start_value = self.get_clock_division()
+
+        # Pre-calculate noise wave shape to save resources
+        self.noise_wave_shape = []
+        for i in range(0, 14):
+            self.noise_wave_shape.append(randint(23, 31))
```

Comment:
[question] It doesn't appear that this is being used. Do you plan on making more changes here?